designer/client/src/components/graph/node-modal/node/FragmentContent.tsx

```diff
@@ -1,39 +1,45 @@
-import React, { useEffect, useState } from "react";
+import React, { useCallback, useState } from "react";
 import { useSelector } from "react-redux";
 import HttpService from "../../../../http/HttpService";
 import { getProcessCounts } from "../../../../reducers/selectors/graph";
 import { FragmentNodeType, Process } from "../../../../types";
 import ErrorBoundary from "../../../common/ErrorBoundary";
 import NodeUtils from "../../NodeUtils";
-import { fragmentGraph as BareGraph } from "../../fragmentGraph";
+import { FragmentGraphPreview } from "../../fragmentGraph";
 import { correctFetchedDetails } from "../../../../reducers/graph/correctFetchedDetails";
 import { getProcessDefinitionData } from "../../../../reducers/selectors/settings";
+import { getFragmentNodesPrefix, useModalDetailsIfNeeded } from "../../../../containers/hooks/useModalDetailsIfNeeded";
+import { usePromiseEffect } from "../../../../containers/hooks/usePromiseEffect";
 
 export function FragmentContent({ nodeToDisplay }: { nodeToDisplay: FragmentNodeType }): JSX.Element {
     const processCounts = useSelector(getProcessCounts);
     const processDefinitionData = useSelector(getProcessDefinitionData);
 
     const [fragmentContent, setFragmentContent] = useState<Process>(null);
+    const { openFragmentNodes } = useModalDetailsIfNeeded();
+
+    usePromiseEffect(
```

Comment:
If I understand correctly what we want to achieve here is:
- set process details
- open fragment node and node of this fragment if it exists. I'm not sure if `openFragmentNodes` describes it correctly. To me, it sounds like there are multiple fragments to open. Maybe, something like openFragmentAndHisNode will be a better name? 
- Do it only once on init? Not sure about this one, but I don't see other cases

WDYT about changing this usePromiseEffect additional abstraction to something like

```
  const processDetailsInitialized = useRef(false);

    useEffect(() => {
        if (fragmentContent) return;
        if (!NodeUtils.nodeIsFragment(nodeToDisplay)) return;
        if (processDetailsInitialized.current) return;

        const handleFragmentContent = async () => {
            const id = nodeToDisplay?.ref.id;
            const { data } = await HttpService.fetchProcessDetails(id);
            const fetchedProcessDetails = correctFetchedDetails(data, processDefinitionData);
            setFragmentContent(fetchedProcessDetails.json);
            return fetchedProcessDetails;
        };

        handleFragmentContent().then((fetchedProcessDetails) => {
            openFragmentNodes(fetchedProcessDetails.json);
        });

        processDetailsInitialized.current = true;
    }, [fragmentContent, nodeToDisplay, openFragmentNodes, processDefinitionData]);
```
?

We will remove additional usePromiseEffect abstraction, which  according to the name sound more like https://gist.github.com/finom/f97a6a4f344690a2486668bb3e207e38